src/http.cc

```diff
@@ -82,9 +82,7 @@ HttpStateData::HttpStateData(FwdState *theFwdState) :
 {
     debugs(11,5, "HttpStateData " << this << " created");
     serverConnection = fwd->serverConnection();
-
-    if (fwd->serverConnection() != nullptr)
-        _peer = cbdataReference(fwd->serverConnection()->getPeer());         /* might be NULL */
+    const auto _peer = peer();
```

Comment:
I would keep that useful reminder unless we are explicitly asked to remove it.

```suggestion
    const auto _peer = peer(); // may be nil
```